pkg/provider-local/controller/controlplane/add.go

```diff
@@ -32,7 +32,11 @@ var (
 	logger = log.Log.WithName("local-controlplane-controller")
 
 	// DefaultAddOptions are the default AddOptions for AddToManager.
-	DefaultAddOptions = AddOptions{}
+	DefaultAddOptions = AddOptions{
+		Controller: controller.Options{
+			RecoverPanic: true,
```

Comment:
Setting this here means that all extensions need to be adapted as well.
Can't we set this rather in https://github.com/gardener/gardener/blob/1190b9321d944da51d90af2a3adc4ca55840d803/extensions/pkg/controller/controlplane/controller.go#L64
to prevent this additional work (at least for the controllers in `extensions/pkg/controller`)?

Reply:
Yes for extension it can be done here, but other changes are fine. Cause we are setting `RecoverPanic: true` as the default.

Reply:
PTAL f3f252f1eaaf9dbd8d91f6df66492dcf4c60a135

Reply:
Why did you only do this for the `ControlPlane` controller? What about the others?

Reply:
Hi, It's not intended, I thought these are all.

Reply:
@rfranzke I think we should not be setting `RecoverPanic` for extensions controllers in g/g [here](https://github.com/gardener/gardener/blob/1190b9321d944da51d90af2a3adc4ca55840d803/extensions/pkg/controller/controlplane/controller.go#L64).
This kind of destroys the idea of extension. Suppose if one sets the value of `RecoverPanic` to false in `provider-extension` then this will not be respected and will be overwritten by the g/g `RecoverPanic` value. According to me, this should not be the case. WDYT?

Reply:
> According to me, this should not be the case. WDYT?

Probably not, you're right. What about respecting the configuration for `RecoverPanic` that could potentially provided by extensions, and if there is nothing then we default it to `true`? Is this possible?

Reply:
I checked it seems we can't, in g/g there is only this https://github.com/gardener/gardener/blob/45624f74d44849822c6bab3c6bbc22f2c70c2f2c/extensions/pkg/controller/backupbucket/controller.go#L64

for adding the extension. And we can't set any such condition which checks if this value was not set. Because when the function call will reach here golang will set its(RecoverPanic) default value to `false`. There was a possibility if this value was a pointer. Please suggest if anything else can be done.

Reply:
Right. Then again, I'm asking my self whether we should still default it to `true` (without option for extensions to opt out) given that it is probably very reasonable to not panic and crash-loop since this would prevent all resources from being reconciled just because one "culprit resource" causing the panic. Do you have a justifyable situation in mind where this is undesired?

TL;DR: My personal opinion is that setting `RecoverPanic=true` still makes sense for extensions. Folks can still add the controller "manually" to their manager if they don't want this default behaviour from the library.

Reply:
As said:

>  Folks can still add the controller "manually" to their manager if they don't want this default behaviour from the library.

(instead of calling the `AddToManager` function)

Reply:
Thanks, Updated it 2de86314144863bb705e8a07b746159143b02acc